### PR TITLE
Timing

### PR DESCRIPTION
**Pull request by mb-dbc-dk**
Added dry-run to replayer
Simplified delay calculation in replayer

### CODE REVIEW

File: recorder/src/main/java/dk/dbc/service/performance/recorder/Config.java [hunk 1]
```diff
@@ -153,7 +152,9 @@ public long getDuration() {
         return duration;
     }
 
-    public long getRunduration() { return runduration; }
+    public long getRunduration() {
+        return runduration;
```
Comment (noah-dbc) [inline recorder/src/main/java/dk/dbc/service/performance/recorder/Config.java]: maybe we could call it runDuration (and getRunDuration) for readability

File: replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java [hunk 2]
```diff
@@ -105,6 +107,12 @@ private static Options options() {
                 .argName("REPLAY")
                 .desc("Replayspeed (ex. 110 is 10% faster than original speed, 0 means no delay between calls")
```
Comment (noah-dbc) [inline replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java:111]: should this documentation change?

File: recorder/src/main/java/dk/dbc/service/performance/recorder/Config.java [hunk 3]
```diff
@@ -259,6 +222,27 @@ private Config(Arguments args, Iterator<String> positionalArguments) throws Pars
         this.javascript = args.take("j", null, t -> t);
     }
 
+    private static long parseTimeSpec(String t) throws RuntimeException {
+        String[] parts = t.split("(?=[^0-9])", 2);
+        if (parts.length != 2)
+            throw new RuntimeException();
+        long number = Long.parseUnsignedLong(parts[0]);
+        if (number < 1)
+            throw new RuntimeException();
```
Comment (noah-dbc) [inline recorder/src/main/java/dk/dbc/service/performance/recorder/Config.java]: maybe IllegalArgumentException or ParseException?

File: replayer/src/main/java/dk/dbc/service/performance/replayer/ReplayerTask.java [hunk 4]
```diff
@@ -43,15 +46,18 @@ public ReplayerTask(Config config, LogCollector logCollector, CallTimeWathcer wa
...
+        this.dryRun = config.isDryRun();
     }
 
     @Override
     public void run() {
-        log.debug( "Running: logLine=" + logLine);
-        long duration = sender.send(logLine, logEntry);
+        log.debug("Running: logLine={}{}", logLine, dryRun ? " (dry-run)" : "");
+        long duration = 0;
+        if (!dryRun)
```
Comment (noah-dbc) [inline replayer/src/main/java/dk/dbc/service/performance/replayer/ReplayerTask.java:56]: I'm a really big fan of using curly braces, even if there is only one line in an if-branch...

File: replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java [hunk 5]
```diff
@@ -205,14 +218,14 @@ public String toString() {
...
-                put( "output", output);
-                put( "replay", String.valueOf(replay));
+                put("durationConstraint", String.valueOf(durationConstraint));
+                put("replayTime", String.valueOf(replayTime));
+                put("callConstraint", String.valueOf(callTimeConstraint) + "/" + maxDelayedCalls + "/" + callBufferSize);
+                put("limit", String.valueOf(limit));
+                put("service", service);
+                put("input", input);
+                put("output", output);
+                put("replay", String.valueOf(replay));
```
Comment (noah-dbc) [inline replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java]: put dry run parameter in here?

**Review by noah-dbc — CHANGES_REQUESTED** [review]

File: replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java (hunk 2 above)
Comment (mb-dbc-dk) [reply replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java:111]: Nope 0 is special case

Comment (noah-dbc) [reply replayer/src/main/java/dk/dbc/service/performance/replayer/Config.java:111]: OK, this still has to be "110" for example, but then you are told in toMap/toString that "replay" is 100/110. Some might become confused that this parameter and the variable are named the same...

**Review by noah-dbc — APPROVED** [review]

**Pull request merged**